src/main/java/Duke.java

```diff
@@ -1,10 +1,275 @@
+    private static int index = 0;
+
     public static void main(String[] args) {
-        String logo = " ____        _        \n"
-                + "|  _ \\ _   _| | _____ \n"
-                + "| | | | | | | |/ / _ \\\n"
-                + "| |_| | |_| |   <  __/\n"
-                + "|____/ \\__,_|_|\\_\\___|\n";
-        System.out.println("Hello from\n" + logo);
+        printGreeting();
+        processRequest();
+    }
+
+    private static void processRequest() {
+        getRequest();
+    }
+
+    private static void getRequest() {
+        Scanner in = new Scanner(System.in);
+        String input;
+        boolean hasExited = false;
+        while (!hasExited) {
+            input = in.nextLine();
+            input = input.trim();
+            hasExited = filterRequest(input);
+        }
+    }
+
+    private static boolean filterRequest(String input) {
+        if (isBye(input)) {
+            printGoodbye();
+            return true;
+        }
+        if (isList(input)) {
+            printList();
+        } else {
+            addOrMarkItem(input);
+        }
+        return false;
+    }
+
+    private static void addOrMarkItem(String input) {
+        String[] words = input.split(SPACEBAR);
+        if (isMark(words)) {
+            markItem(words);
+        } else if (isUnmark(words)) {
+            unmarkItem(words);
+        } else {
+            addTask(input);
+        }
+    }
+
+    private static void markItem(String[] words) {
+        int indexToMark = getIndex(words);
+        if (isIndexValid(indexToMark)) {
+            list[indexToMark].markAsDone();
+            printMarkIsCompleted(list[indexToMark]);
+        } else {
+            handleInvalidIndex();
+        }
+    }
+
+    private static void unmarkItem(String[] words) {
+        int indexToUnmark = getIndex(words);
+        if (isIndexValid(indexToUnmark)) {
+            list[indexToUnmark].markAsUndone();
+            printUnmarkIsCompleted(list[indexToUnmark]);
+        } else {
+            handleInvalidIndex();
+        }
+    }
+
+    private static void handleInvalidIndex() {
+        System.out.println(INVALID_INDEX);
+        System.out.println(LINE);
+    }
+
+    private static boolean isIndexValid(int indexToUnmark) {
+        boolean isAboveZero = indexToUnmark >= MINIMUM_INDEX;
+        boolean isBelowMaximum = indexToUnmark < index;
+        return isAboveZero && isBelowMaximum;
+    }
+
+    private static int getIndex(String[] words) {
+        return Integer.parseInt(words[1]) - 1;
+    }
+
+    private static void addTask(String input) {
+        String typeOfTask = getTypeOfTask(input);
+        if (isToDo(typeOfTask)) {
+            addToDo(input);
+        } else if (isDeadline(typeOfTask)) {
+            addDeadline(input);
+        } else if (isEvent(typeOfTask)) {
+            addEvent(input);
+        } else {
+            addNormalTask(input);
+        }
+        printConfirmation();
+        incrementIndex();
+    }
```

Comment:
As with all other methods that you use to process the user's inputs, I think it might be easier if switch-case statements were used instead so that there isn't a need to keep reading the user's input again and again? Another suggestion would be to have these methods in a separate Parser class that deals with breaking down and processing the user's input.